tools/autograd/templates/Functions.cpp

```diff
@@ -439,6 +443,14 @@ std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<
     auto size = shape[dim];
     accumulate += size;
     grad_inputs[i] = grad.narrow(dim, accumulate - size, size);
+    if (pad) {
+      for (size_t j = 0; j < sizes.size(); ++j) {
```

Comment:
I'm confused by this loop (but maybe I don't understand what happens if inputs have different dimensionality).  `cursize` is iterating over `shape`, but the end of the loop is `sizes.size()`, i.e. the number of input shapes.  How can that matter for grad_input[i]?

Reply:
If all inputs have same dimensionality except the cat dim, we just need to narrow along cat dim in order to get the real grad input. However, if all inputs have different dim, we need narrow every dim to get the real grad input.

Reply:
I got what you mean now. It should be shape.size() instead of sizes.size(). I will correct it.